pandas/tests/generic/methods/test_reorder_levels.py

```diff
@@ -0,0 +1,76 @@
+import numpy as np
+import pytest
+
+from pandas import DataFrame, MultiIndex, Series
+import pandas._testing as tm
+
+
+class TestReorderLevels:
+    @pytest.mark.parametrize("klass", [Series, DataFrame])
+    def test_reorder_levels(self, klass):
+        index = MultiIndex(
+            levels=[["bar"], ["one", "two", "three"], [0, 1]],
+            codes=[[0, 0, 0, 0, 0, 0], [0, 1, 2, 0, 1, 2], [0, 1, 0, 1, 0, 1]],
+            names=["L0", "L1", "L2"],
+        )
+        df = DataFrame({"A": np.arange(6), "B": np.arange(6)}, index=index)
+        obj = df if klass is DataFrame else df["A"]
+
+        # no change, position
+        result = obj.reorder_levels([0, 1, 2])
+        tm.assert_equal(obj, result)
```

Comment:
if this is going to happen a lot with these changes, should maybe consider a solution that can maintain the correct assertion helper.

either in the parameterisation as "klass, helper" or create a fixture something like

```
@fixture()
def assert_klass_equal(klass):
    if isinstance(klass, DataFrame):
      return tm.assert_frame_equal
    elif isinstance(klass, Series):
      return tm.assert_series_equal
   else:
      assert False 
```
(code not tested!)

Reply:
might make sense to add as a kwarg to tm.assert_equal

Reply:
yeah, that sounds good as well. could be follow-up